samples/webcodecs-echo/js/stream_worker.js

```diff
@@ -0,0 +1,685 @@
+         });
+         VideoEncoder.isConfigSupported(config).then((encoderSupport) => {
+           if(encoderSupport.supported) {
+             this.encoder.configure(encoderSupport.config);
+             self.postMessage({text: 'Encoder successfully configured:\n' + JSON.stringify(encoderSupport.config)});
+             // self.postMessage({text: 'Encoder state: ' + JSON.stringify(this.encoder.state)});
+           } else {
+             self.postMessage({severity: 'fatal', text: 'Config not supported:\n' + JSON.stringify(encoderSupport.config)});
+           }
+         })
+         .catch((e) => {
+            self.postMessage({severity: 'fatal', text: `Configuration error: ${e.message}`});
+         })
+       },
+       transform(frame, controller) {
+         if (this.pending_outputs <= 30) {
+           this.pending_outputs++;
+           const insert_keyframe = (this.frameCounter % config.keyInterval) == 0;
+           this.frameCounter++;
+           try {
+             if (this.encoder.state != "closed") {
+               if (this.frameCounter % 20 == 0) {
+                 // self.postMessage({text: 'Encoded 20 frames'});
+               }
+               const queue = this.encoder.encodeQueueSize;
+               encqueue_update(queue);
+               const before = performance.now();
+               this.encoder.encode(frame, { keyFrame: insert_keyframe });
+               const after = performance.now();
+               const duration = after - before;
+               enc_update(duration);
+             } 
+           } catch(e) {
+             self.postMessage({severity: 'fatal', text: `Encoder Error: ${e.message}`});
+           }
+         }
+         frame.close();
+       }
+     });
+   }
+
+   stop() {
+     const enc_stats = enc_report();
+     const encqueue_stats = encqueue_report();
+     const dec_stats = dec_report();
+     const decqueue_stats = decqueue_report();
+     const rtt_stats = rtt_report();
+     self.postMessage({severity: 'info', text: 'RTT report: ' + JSON.stringify(rtt_stats)});  
+     self.postMessage({severity: 'info', text: 'Encoder Time report: ' + JSON.stringify(enc_stats)});
+     self.postMessage({severity: 'info', text: 'Encoder Queue report: ' + JSON.stringify(encqueue_stats)});
+     self.postMessage({security: 'info', text: 'Decoder Time report: ' + JSON.stringify(dec_stats)});
+     self.postMessage({severity: 'info', text: 'Decoder Queue report: ' + JSON.stringify(decqueue_stats)});
+     if (stopped) return;
+     stopped = true;
+     this.stopped = true;
+     self.postMessage({severity: 'fatal', text: 'stop() called'});
+     // TODO: There might be a more elegant way of closing a stream, or other
+     // events to listen for.
+     if (encoder.state != "closed") encoder.close();
+     if (decoder.state != "closed") decoder.close();
+     self.postMessage({severity: 'fatal', text: "stop(): frame, encoder and decoder closed"});
+     return;
+   }
+
+   SendStream(self, transport) {
```

Comment:
I like this pattern, but it's an odd upper-case named function which call sites aren't using as a constructor (e.g. `new this.SendStream(...)`), so how about:
```suggestion
   createSendStream(self, transport) {
```
?

Reply:
Fixed.